docs/tutorials/04_ParticleFilter.py

```diff
@@ -0,0 +1,211 @@
+#
+# One significant drawback of this is that we are going to be sending a lot of particles through
+# multiple steps, resulting in a decent computational complexity.
+# Also, we must be careful that we select an appropriate particle/sample sparsity to get a good
+# description of the prior distribution.
+# After a few iterations, most particles will have moved far enough away that they have
+# :math:`\sim 0` weight, hence provide no contribution to our estimates. But they are still being
+# processed through at each predict and update step, which is quite a computational waste. Hence a
+# re-sampling method would be useful to apply every so often (many exist, and are designed to
+# redistribute these particles to areas where the posterior probability is higher).
+
+# %%
+
+from datetime import datetime
+from datetime import timedelta
+
+from stonesoup.models.transition.linear import CombinedLinearGaussianTransitionModel, \
+    ConstantVelocity
+from stonesoup.types.groundtruth import GroundTruthPath, GroundTruthState
+
+start_time = datetime.now()
+transition_model = CombinedLinearGaussianTransitionModel([ConstantVelocity(0.05),
+                                                          ConstantVelocity(0.05)])
+
+truth = GroundTruthPath([GroundTruthState([0, 1, 0, 1], timestamp=start_time)])
+
+for k in range(1, 21):
+    truth.append(GroundTruthState(
+        transition_model.function(truth[k-1], noise=True, time_interval=timedelta(seconds=1)),
+        timestamp=start_time+timedelta(seconds=k)))
+
+# %%
+
+# Plot the ground truth.
+
+from matplotlib import pyplot as plt
+
+fig = plt.figure(figsize=(10, 6))
+ax = fig.add_subplot()
+ax.set_xlabel("$x$")
+ax.set_ylabel("$y$")
+ax.axis('equal')
+
+ax.plot([state.state_vector[0] for state in truth],
+        [state.state_vector[2] for state in truth],
+        linestyle="--")
+
+# %%
+# Create sensor with bearing, range measurement model.
+
+import numpy as np
+
+from stonesoup.models.measurement.nonlinear import CartesianToBearingRange
+from stonesoup.types.detection import Detection
+
+sensor_x = 0
+sensor_y = 0
+
+measurement_model = CartesianToBearingRange(
+    ndim_state=4,
+    mapping=(0, 2),
+    noise_covar=np.diag([np.radians(0.1), 0.1]),  # bad bearing, good range uncertainties
+    translation_offset=np.array([[sensor_x], [sensor_y]])
+)
+
+measurements = []
+for state in truth:
+    measurement = measurement_model.function(state, noise=True)
+    measurements.append(Detection(measurement, timestamp=state.timestamp))
+
+# %%
+# Plot detections.
+
+from stonesoup.functions import pol2cart
+
+x, y = pol2cart(
+    np.hstack(state.state_vector[1, 0] for state in measurements),
+    np.hstack(state.state_vector[0, 0] for state in measurements))
+ax.scatter(x + sensor_x, y + sensor_y, color='b')
+fig
+
+# %%
+# Create a :class:`~.ParticlePredictor`, which takes a collection of particles, with a given
+# transition model, and predicts them forward as with the :class:`~.KalmanPredictor`.
```

Comment:
It may be worth rewording this slightly, as I had to read again, as I thought it implied the particles were predicted using the `KalmanPredictor`